### PR TITLE
Proxy support using proxyplease

### PR DESCRIPTION
**Pull request by Sebbb**
https://github.com/aus/proxyplease enables proxy support for Linux and Windows, including SSO

### CODE REVIEW

**Comment by Sebbb** [comment]
Google CLA signed, please rescan. Thanks.

**Comment by ThomasHabets** [comment]
Sorry for the delay. I'll look at this in a few days.

File: huproxyclient/client.go [hunk 1]
```diff
@@ -89,7 +90,15 @@ func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	dialer := websocket.Dialer{}
+	dialContext := proxyplease.NewDialContext(proxyplease.Proxy{})
+
+	dialer := websocket.Dialer{
+		ReadBufferSize:   1024,
```
Comment (ThomasHabets) [inline huproxyclient/client.go:96]: Why also set these buffer sizes?

File: huproxyclient/client.go [hunk 2]
```diff
@@ -89,7 +90,15 @@ func main() {
...
+		WriteBufferSize:  1024,
+		HandshakeTimeout: 45 * time.Second,
```
Comment (ThomasHabets) [inline huproxyclient/client.go:98]: Add handshake timeout should probably be its own commit.

**Review by ThomasHabets — CHANGES_REQUESTED** [review]

File: huproxyclient/client.go (hunk 1 above)
Comment (Sebbb) [reply huproxyclient/client.go:96]: I took it over from the example using Websockets - https://github.com/aus/proxyplease/blob/master/example/websocket.go
I don't know why they took these values, also not what values would be reasonable or if it makes sense to set them at all.
So, I could remove these settings, verify that it works for me with and without proxy, but that's it..

Comment (ThomasHabets) [reply huproxyclient/client.go:96]: Please do.

**Comment by x-077** [comment]
Hello, 

do you plan to add this feature ?

**Comment by ThomasHabets** [comment]
Sorry, I forgot about this. I was waiting for the PR author to update according to comments. I'll give them a day and will then merge and do the changes myself.

**Comment by x-077** [comment]
Hello @ThomasHabets , 

no worries, I saw that someone forked and implemented it : https://github.com/dono/huproxy and was waiting to see how you implement it to compare.

**Comment by ThomasHabets** [comment]
Merged outside this PR.
